core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -167,7 +256,275 @@ public void setTransportSender(TransportSender transportSender) {
+        Method method = new EndpointDispatcher().getOnPongMessageMethod(webSocketEndpoint);
+        if (method != null) {
+            List<Object> parameterList = new LinkedList<>();
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == PongMessage.class) {
+                            parameterList.add(new WebSocketPongMessage(controlCarbonMessage.readBytes()));
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters " +
+                                                                           "without @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
+                        }
+                    }
+            );
+        }
+    }
+
+    private void handleError(CarbonMessage carbonMessage, Throwable throwable,
+                             PatternPathRouter.RoutableDestination<Object> routableEndpoint,
+                             Session session) {
+        Object webSocketEndpoint = routableEndpoint.getDestination();
+        Map<String, String> paramValues = routableEndpoint.getGroupNameValues();
+        Method method = new EndpointDispatcher().getOnErrorMethod(webSocketEndpoint);
+        if (method != null) {
+            List<Object> parameterList = new LinkedList<>();
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == Throwable.class) {
+                            parameterList.add(throwable);
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters " +
+                                                                           "without @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
```

Comment:
IMO shouldn't we add new Object() here to avoid possible null pointer exceptions. Consider using new Object()